generated/nifgen/session.py

```diff
@@ -1969,6 +1969,31 @@ def delete_script(self, script_name):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    @ivi_synchronized
+    def delete_waveform(self, waveform_name_or_handle):
+        '''delete_waveform
+
+        Removes a previously created arbitrary waveform from the signal generator memory and invalidates the waveform handle.
+
+        Note: The signal generator must not be in the Generating state when you call this method.
+
+        Tip:
+        This method requires repeated capabilities (usually channels). If called directly on the
+        nifgen.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        nifgen.Session instance, and calling this method on the result.:
+
+            session.channels['0,1'].delete_waveform(waveform_name_or_handle)
+
+        Args:
+            waveform_name_or_handle (str or int): The name (str) or handle (int) of an arbitrary waveform previously allocated with allocate_named_waveform or allocate_waveform.
+
+        '''
+        if isinstance(waveform_name_or_handle, str):
```

Comment:
Why not try using `try`/`except`? We've learned it's more Pythonic.
ctypes will complain, is my guess, based on the order in which you try.

Reply:
Do I change write_waveform too?